spec/integration/api/posts_controller_spec.rb

```diff
@@ -215,36 +191,51 @@
         post(
           api_v1_posts_path,
           params: {
-            access_token: access_token_with_read_and_write,
+            access_token: access_token,
             body:         "Hello this is a public post!",
             public:       true
           }
         )
         expect(response.status).to eq(200)
         post = response_body(response)
-        confirm_post_format(post, auth_with_read_and_write.user, post_for_ref_only)
+        confirm_post_format(post, auth.user, post_for_ref_only)
       end
 
       it "or creates a private post" do
-        aspect = Aspect.find_by(user_id: auth_with_read_and_write.user.id)
-        post_for_ref_only = auth_with_read_and_write.user.post(
+        aspect = auth.user.aspects.create(name: "new aspect")
+        post_for_ref_only = auth.user.post(
           :status_message,
           text:       "Hello this is a private post!",
-          aspect_ids: [aspect[:id]]
+          aspect_ids: [aspect.id]
         )
 
         post(
           api_v1_posts_path,
           params: {
-            access_token: access_token_with_read_and_write,
+            access_token: access_token,
             body:         "Hello this is a private post!",
             public:       false,
-            aspects:      [aspect[:id]]
+            aspects:      [aspect.id]
           }
         )
         post = response_body(response)
         expect(response.status).to eq(200)
-        confirm_post_format(post, auth_with_read_and_write.user, post_for_ref_only)
+        confirm_post_format(post, auth.user, post_for_ref_only)
+      end
+
+      it "doesn't creates a private post without private:modify scope in token" do
+        aspect = auth.user.aspects.create(name: "new aspect")
+        post(
+          api_v1_posts_path,
+          params: {
+            access_token: access_token_public_only,
+            body:         "Hello this is a private post!",
+            public:       false,
+            aspects:      [aspect.id]
+          }
+        )
+
+        expect(response.status).to eq(422)
```

Comment:
403 seems a better error code.

Reply:
The Post API documentation had this as 403 but with a parenthetical note that it should be changed to 422 so I opted to do that.